### PR TITLE
Implementation of set_delegate_voter

### PR DESCRIPTION
**Pull request by ishaan-jaff**

### CODE REVIEW

File: tests/locked-voter.spec.ts [hunk 1]
```diff
@@ -250,6 +250,38 @@ describe("Locked Voter", () => {
...
+  
+      
+    it("Set vote delegate access control test", async () => {
+      const expectedDelegate = Keypair.generate().publicKey;
+      const incorrectPubKey = Keypair.generate().publicKey;
+      const tx = await lockerW.setVoteDelegate(
+        expectedDelegate,
+        incorrectPubKey
+      );
+      tx.addSigners(user);
```
Comment (ishaan-jaff) [inline tests/locked-voter.spec.ts:275]: is an access control test where there is an incorrect signer ?

**Comment by michaelhly** [comment]
@ishaan-jaff would you mind running `cargo fmt`. 🙏

File: programs/locked-voter/src/lib.rs [hunk 2]
```diff
@@ -72,6 +72,12 @@ pub mod locked_voter {
         ctx.accounts.cast_vote(side)
     }
 
+    /// Delegate escrow vote
```
Comment (michaelhly) [inline programs/locked-voter/src/lib.rs:75]: nit
```suggestion
    /// Delegate escrow vote.
```

File: tests/locked-voter.spec.ts [hunk 3]
```diff
@@ -251,6 +251,30 @@ describe("Locked Voter", () => {
...
+
+    it("Set vote delegate access control test", async () => {
+      const expectedDelegate = Keypair.generate().publicKey;
+      const incorrectAccount = Keypair.generate();
+      const tx = await lockerW.setVoteDelegate(
+        expectedDelegate,
+        incorrectAccount.publicKey
+      );
+      tx.addSigners(incorrectAccount);
+      await expectTX(tx).to.be.rejectedWith(/0xbc4/);
```
Comment (michaelhly) [inline tests/locked-voter.spec.ts:275]: I don't think this is the error code we're looking for ...

File: tests/locked-voter.spec.ts [hunk 4]
```diff
@@ -251,6 +251,35 @@ describe("Locked Voter", () => {
...
+      const tx = new TransactionEnvelope(lockerW.sdk.provider, [
+        lockerW.program.instruction.setVoteDelegate(expectedDelegate, {
+          accounts: {
+            escrow,
+            escrowOwner: incorrectAccount.publicKey,
+          },
+        }),
+      ]);
+      tx.addSigners(incorrectAccount);
+      await expectTX(tx).to.be.rejectedWith(/0x44c/);
```
Comment (ishaan-jaff) [inline tests/locked-voter.spec.ts:280]: we get the following log with the updated test (this looks like an access control error) 
` 'Program log: Instruction: SetVoteDelegate',
      'Program log: KeyMismatch: Keys do not match.',
      'Program log: self.escrow.owner != self.escrow_owner',
       Custom program error: 0x44c
`

**Review by michaelhly — APPROVED** [review]

**Pull request merged**